### PR TITLE
Allow Custom Loss function inheriting from _Loss

### PR DESCRIPTION
**Pull request by nishai**
PR for Issue #397 

* Replaced check for existing `torch.nn.modules.loss` method, with a check for if the custom loss is a subclass of `torch.nn.modules.loss._Loss`.
* Added a test case to verify whether a custom loss extending `torch.nn.modules.loss._Loss` works.

### CODE REVIEW

**Comment by ourownstory** [comment]
Looks good. will merge into a branch and run tests on my system, too.
